src/App.js

```diff
@@ -47,22 +66,48 @@ const App = () => {
     //    3 squares in each column match
     // 3. Go across each diagonal to see if 
     //    all three squares have the same value.
+    
+    //check if winner is in rows else they are in columns:
+    for (let i = 0; i < 3; i++) {
+      if (squares[i][0].value === squares[i][1].value && squares[i][1].value === squares[i][2].value && squares[i][0].value !== '') {
+        return squares[i][0].value;
+      } else if (squares[0][i].value === squares[1][i].value && squares[1][i].value === squares[2][i].value && squares [0][i].value !== '') {
+        return squares[0][i].value;
+      } 
+    }
+    //winner in both diagonals:
+    if (squares[0][0].value === squares[1][1].value && squares[1][1].value === squares[2][2].value & squares[0][0].value !== '') {
+      return squares[0][0].value;
+    } else if (squares[0][2].value === squares[1][1].value && squares[1][1].value === squares[2][0].value && squares[0][0].value !== '') {
+      return squares[0][2].value;
+    } 
 
-  }
+    for (let r = 0; r < 3; r++) {
+      for (let c = 0; c < 3; c++) {
+        if (squares[r][c].value === '') {
+          return null;
+        }
+      }}
+      //returning null if there are empty squares
+      return 'Tie';
+
+  };
 
-  const resetGame = () => {
+  function reset() {
     // Complete in Wave 4
+    setSquares(generateSquares())
```

Comment:
This is a great start!  The only pieces that are needed are resetting the player and the winner.
```suggestion
    setSquares(generateSquares())
    setCurrentPlayer(PLAYER_1)
    setWinner(null)
```